cli/launchpad/generic.go

```diff
@@ -0,0 +1,106 @@
+package launchpad
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+)
+
+const indentSize = 2
+
+var (
+	errValidationFailed = errors.New("validation failed")
+	tfNameRegex         = regexp.MustCompile("^[a-zA-Z][a-zA-Z\\d\\-\\_]*$")
+)
+
+//
+type resourcer interface {
+	// validate ensures the parsed YAML has validate parameters.
+	validate() error
+	// kind returns the validated crdKind of the resource.
+	kind() crdKind
+	// enroll adds the resource into final eval mapping.
+	enroll(*eval) error
+}
+
+type resourcespecer interface {
```

Comment:
Having this separate interface feels wrong to me. I'd see if we can consolidate everything into the top-level interface.